packages/replayio/src/utils/recordings/upload/uploadRecording.ts

```diff
@@ -91,19 +92,22 @@ export async function uploadRecording(
           debug(`Attempt ${attemptNumber} to set metadata failed:\n%j`, error);
         }
       );
-
-      await retryWithExponentialBackoff(
-        () =>
-          uploadRecordingFile({
-            recordingPath: path,
-            size,
-            uploadLink,
-          }),
-        (error: unknown, attemptNumber: number) => {
-          debug(`Attempt ${attemptNumber} to upload failed:\n%j`, error);
-        }
+      await uploadQueue.add(() =>
+        retryWithExponentialBackoff(
+          () =>
+            uploadRecordingFile({
+              recordingPath: path,
+              size,
+              url: uploadLink,
+            }),
+          (error: any, attemptNumber: number) => {
+            debug(`Attempt ${attemptNumber} to upload failed:\n%j`, error);
+            if (error.code === "ENOENT") {
```

Comment:
It's quite an edge case and it shouldn't happen - but it did happen before https://github.com/replayio/replay-cli/pull/364 . Either way, it's worth handling it here. If the target file doesn't exist there is no point in retrying (throwing in this `onFail` callback prevents the next retrying attempt).